{% extends "base/plain.html" %}

{% load booking %}

{% block content %}
<style>
  .image-header-gvl {
    -webkit-column-count: 3;
    -moz-column-count: 3;
    column-count: 3;

    -webkit-column-gap: 140px;
    -moz-column-gap: 140px;
    column-gap: 140px;
  }
</style>

<div class="image-header-gvl">
    <a href=http://genome.edu.au><img src="../media/gvl_logo.jpg" alt="GVL logo" width="112"/></a>
    <a href="http://galaxyproject.org"><img src="../media/GalaxyLogoHighRes.png" width="164" /></a>
    <a href=http://vlsci.org.au><img src="../media/vlsci_logo.jpg" alt="VLSCI logo" width="164"/></a>
</div>

<h1 id="introduction-to-galaxy">Introduction to Galaxy</h1>
<p>Written and maintained by <a href="mailto:simon.gladman@unimelb.edu.au">Simon Gladman</a> - VLSCI</p>
<h2 id="background">Background</h2>
<p>Galaxy is a web based analysis and workflow platform designed for biologists to analyse their own data. It comes with most of the popular bioinformatics tools already installed and ready for use. There are many Galaxy servers around the world and some are tailored with specific toolsets and reference data for analysis of human genomics, microbial genomics, proteomics etc.</p>
<p>There are some introductory slides available <a href="https://docs.google.com/presentation/d/1dzHagGkswjH7MOZ7OACVXGU-riBs33K3J5lWpnCpPhs/pub?start=false&amp;loop=false&amp;slide=id.g537781ff1_2_18">here</a>.</p>
<p>Basically, the Galaxy interface is separated into 3 parts. The tool list on the left, the viewing pane in the middle and the analysis and data history on the right. We will be looking at all 3 parts in this tutorial.</p>
<p><img src="../media/Galaxy_components.png" alt="Galaxy interface components diagram" width="70%" /></p>
<p>This workshop/tutorial will familiarize you with the Galaxy interface. It will cover the following topics:</p>
<ul>
<li>Logging in to the server</li>
<li>Getting data into galaxy</li>
<li>How to access the tools</li>
<li>Using to use some common tools</li>
</ul>
<hr />
<h2 id="learning-objectives">Learning Objectives</h2>
<p>At the end of this tutorial you should:</p>
<ol>
<li>Be able to register on and login to a Galaxy server.</li>
<li>Be able to upload data to a Galaxy server from:<ul>
<li>A file on your local computer</li>
<li>A file on a remote datastore with an accessible URL.</li>
</ul>
</li>
<li>Be able use tools in Galaxy by:<ul>
<li>Accessing the tool via the tool menu</li>
<li>Using the tool interface to run the particular tool</li>
<li>Viewing/accessing the tool output.</li>
</ul>
</li>
</ol>
<hr />
<h2 id="section-1-preparation">Section 1: Preparation.</h2>
<p>The purpose of this section is to get you to log in to the server..</p>
<ol>
<li>
<p>Go to the ip address of your GVL Galaxy server (or if you don’t have one, open <a href="http://galaxy-tut.genome.edu.au">Galaxy-Tut</a> server) in Firefox or Chrome (your choice) - Please don’t use Internet Explorer or Safari.</p>
</li>
<li>
<p>If you have previously registered on this server just log in:</p>
<ul>
<li>On the top menu select: <strong>User -&gt; Login</strong></li>
<li>Enter your password</li>
<li>Click <strong>Submit</strong></li>
</ul>
</li>
<li>
<p>If you haven’t registered on this server, you’ll need to now.</p>
<ul>
<li>On the top menu select: <strong>User -&gt; Register</strong></li>
<li>Enter your email, choose a password, repeat it and add a (all lower case) one word name</li>
<li>Click <strong>Submit</strong></li>
</ul>
</li>
</ol>
<hr />
<h2 id="section-2-getting-data-into-galaxy">Section 2: Getting data into Galaxy</h2>
<p>There are 2 main ways to get your data into Galaxy. We will use each of these methods for 3 files and then work use those 3 files for the rest of the workshop.</p>
<ul>
<li>
<p>Start a new history for this workshop. To do this:</p>
<ul>
<li>Click on the history menu button (the <img src="../media/Galaxy-menu.png" width=20 /> icon) at the top of the Histories panel.</li>
<li>Select <strong>Create New</strong></li>
</ul>
</li>
</ul>
<p>It is important to note that Galaxy has the concept of "File Type" built in. This means that each file stored needs to have its type described to Galaxy as it is being made available. Examples of file types are: text, fasta, fastq, vcf, GFF, Genbank, tabular etc.</p>
<p>We will tell Galaxy what type of file each one is as we upload it.</p>
<h3 id="method-1-upload-a-file-from-your-own-computer">Method 1: Upload a file from your own computer</h3>
<p>With this method you can get most of the files on your own computer into Galaxy. (there is a size limit)</p>
<ul>
<li>
<p>Download the following file to your computer: <em>https://swift.rc.nectar.org.au:8888/v1/AUTH_377/public/galaxy101/Contig_stats.txt.gz</em></p>
<ul>
<li>From the Galaxy tool panel, click on <strong>Get Data -&gt; Upload File</strong></li>
<li>Click the <strong>Choose File</strong> button</li>
<li>Find and select the <em>Contig_stats.txt.gz</em> file you downloaded and click <strong>Open</strong></li>
<li>Set the "file format" to <em>tabular</em></li>
<li>Click the <strong>Start</strong> button</li>
<li>Once the progress bar reaches 100%, click the <strong>Close</strong> button</li>
</ul>
</li>
</ul>
<p>The file will now upload to your current history.</p>
<h3 id="method-2-upload-a-file-from-a-url">Method 2: Upload a file from a URL</h3>
<p>If a file exists on a web resource somewhere and you know its URL (Unique resource location - a web address) you can directly load it into Galaxy.</p>
<ul>
<li>
<p>From the tool panel, click on <strong>Get Data -&gt; Upload File</strong></p>
<ul>
<li>Click on the <strong>Paste/Fetch Data</strong> button</li>
<li>Copy and paste the following web address into the URL/Text box: <em>https://swift.rc.nectar.org.au:8888/v1/AUTH_377/public/COMP90014/Assignment1/bacterial_std_err_1.fastq.gz</em></li>
<li>Set the file format to <em>fastqsanger</em> (not fastqcsanger)</li>
<li>Click <strong>Start</strong></li>
<li>Once the progress bar has reached 100%, click <strong>Close</strong></li>
</ul>
</li>
</ul>
<p>Note that Galaxy is smart enough to recognize that this is a compressed file and so it will uncompress it as it loads it.</p>
<h3 id="method-2-again-get-data-from-a-public-url">Method 2 (again): Get data from a public URL</h3>
<p>Now we are going to upload another file from the remote data source.</p>
<p>Repeat the above for: <em>https://swift.rc.nectar.org.au:8888/v1/AUTH_377/public/MRSA0252.fna</em></p>
<p>Note that this file is a <em>fasta</em> file and not a <em>fastqsanger</em> file.</p>
<div class="showable-container details showable-hidden" id="showable0">
<div class="showable-inner">
<div class="showable-header details" id="showabletitle0"><a href="" id="showablelink0">Reveal detailed instructions</a></div>
<div class="showable-body details" id="showablebody0">
<ul>
<li>
<p>From the tool panel, click on <strong>Get Data -&gt; Upload File</strong></p>
<ul>
<li>Click on the <strong>Paste/Fetch Data</strong> button</li>
<li>Copy and paste the following web address into the URL/Text box: https://swift.rc.nectar.org.au:8888/v1/AUTH_377/public/MRSA0252.fna</li>
<li>Set the file format to <em>fasta</em>.</li>
<li>Click <strong>Start</strong></li>
<li>Once the progress bar has reached 100%, click <strong>Close</strong></li>
</ul>
</li>
</ul>
</div>
</div>
</div>
<script>(function(w,d,u){w.readyQ=[];w.bindReadyQ=[];function p(x,y){if(x=="ready"){w.bindReadyQ.push(y);}else{w.readyQ.push(x);}};var a={ready:p,bind:p};w.$=w.jQuery=function(f){if(f===d||f===u){return a}else{p(f)}}})(window,document)</script>
<script>
    $(document).ready(function(){
        $("#showablelink0").click(function(e){
            e.preventDefault();
            $("#showable0").toggleClass("showable-hidden");
        });
    });
    </script>
<p>The DNA sequence of <em>Staphlococcus aureus MRSA252</em> will be loaded into your history as a fasta file.</p>
<p>Your history should now look like this.</p>
<p><img src="../media/starting-history.png" width="300px" align="center" /></p>
<h3 id="the-data">The data</h3>
<p>Though we aren't going to focus on the contents of these files and what they mean from a bioinformatics standpoint, here is a brief description of each one.</p>
<ul>
<li>
<p><em>Contigs_stats.txt</em></p>
<ul>
<li>this file contains a table of summary data from a de novo genome assembly (the process of attempting to recover the full genome of an organism from the short read sequences produced by most DNA sequencing machines. )</li>
<li>The columns contain a lot of information but the ones we will be using indicate the amount of data (or coverage) that went into making up each piece of the final assembly.  </li>
</ul>
</li>
</ul>
<!-- -->

<ul>
<li>
<p><em>bacterial_std_err_1.fastq.gz</em></p>
<ul>
<li>This file contains sequence reads as they would come off an Illunina sequencing machine. They are in <a href="https://en.wikipedia.org/wiki/FASTQ_format">fastq</a> format.  </li>
</ul>
</li>
</ul>
<!-- -->

<ul>
<li>
<p><em>MRSA0252.fna</em></p>
<ul>
<li>This file contains the genome sequence of <em>Staphylococcus aureus MRSA252</em>. It is in <a href="https://en.wikipedia.org/wiki/FASTA_format">fasta</a> format.</li>
</ul>
</li>
</ul>
<hr />
<h2 id="section-3-play-with-the-tools">Section 3: Play with the tools</h2>
<p>The purpose of this section is to get you used to using the available tools in Galaxy and point out some of the more basic manipulation tools.</p>
<p>Firstly however, you’ll notice that two of the files have very long and confusing names. So we might want to change them. To do this we need to “edit” the file. So:</p>
<ol>
<li>Click on the <img src="../media/Galaxy-edit.png" width=20 /> icon (edit) next to the file in the history called: <em>https://swift.rc.nectar.org.au:8888/v1/AUTH_377/public/COMP90014/Assignment1/bacterial_std_err_1.fastq</em></li>
<li>In the "Name" text box, give it a new name. Call it: <em>Typical Fastq File</em></li>
<li>Click the <strong>Save</strong> button.</li>
</ol>
<p>Repeat the process for the MRSA252 fasta file. Rename it to <em>MRSA252.fna</em></p>
<p>Now that’s better. There was a lot of other functionality hidden behind that edit (<img src="../media/Galaxy-edit.png" width=20 />) icon. You can change a file’s data type, convert its format and many other things. Feel free to play around with them at a later date.</p>
<p>Ok, back to the tools..</p>
<h3 id="example-1-histogram-and-summary-statistics">Example 1: Histogram and summary statistics</h3>
<p>The first thing we are going to do is produce a histogram of contig read coverage depths and calculate the summary statistics from the Contig_stats.txt file. To do this we need to cut out a couple of columns, remove a line and then produce a histogram. This will introduce some of the text manipulation tools.</p>
<p>Click on the <img src="../media/Galaxy-view.png" width=20 /> icon of the <em>Contig_stats.txt</em> file to have a look at it. Note that there are 18 columns in this file. We want column 1 and column 6. To do this:</p>
<p><strong>1. Cut out column 1 and column 6.</strong></p>
<ul>
<li>From the tool panel, click on <strong>Text Manipulation -&gt; Cut</strong> and set the following:</li>
<li>Set "Cut Columns" to: <em>c1,c6</em></li>
<li>"Delimited by": <em>Tab</em></li>
<li>"Cut from": <em>Contig_stats.txt</em></li>
<li>Click <strong>Execute</strong></li>
</ul>
<p>Examine the new file by clicking on it’s <img src="../media/Galaxy-view.png" width=20 /> icon. We now have 2 columns instead of the 18 in the original file.</p>
<p><strong>2. Remove the Header lines of the new file.</strong></p>
<ul>
<li>From the tool panel, click on <strong>Text Manipulation -&gt; Remove beginning</strong> and set the following:</li>
<li>"Remove First": <em>1</em></li>
<li>"from": <em>Cut on data1</em></li>
<li>click <strong>Execute</strong></li>
</ul>
<p>Note the the new file is the same as the previous one without the header line.</p>
<p><strong>3. Make a histogram.</strong></p>
<ul>
<li>From the tool panel, click on <strong>Graph/Display Data -&gt; Histogram</strong> and set the following:</li>
<li>"Dataset": <em>Remove beginning on Data X</em></li>
<li>"Numerical column for X axis": <em>c2</em></li>
<li>"Number of breaks": <em>25</em></li>
<li>"Plot title": <em>Histogram of Contig Coverage</em></li>
<li>"Label for X axis": <em>Coverage depth</em></li>
<li>Click <strong>Execute</strong></li>
</ul>
<p>Click on the <img src="../media/Galaxy-view.png" width=20 /> icon of the histogram to have a look at it. Note there are a few peaks.. Maybe these correspond to single, double and triple copy number of these contigs.</p>
<p><strong>4. Calculate summary statistics for contig coverage depth.</strong></p>
<ul>
<li>From the tool panel, click on <strong>Statistics -&gt; Summary Statisitics</strong> and set the following:</li>
<li>"Summary statistics on": <em>Remove beginning on Data X</em></li>
<li>"Column or expression": <em>c2</em></li>
<li>Click <strong>Execute</strong></li>
</ul>
<p>You’ll note that the summary statistics tool failed and is red in the history. There was an error! If you click on the filename, and then the bug <img src="../media/Galaxy-bug.png" width=20 /> symbol, it will tell you what went wrong. (There is a missing python library.) At this point, you would normally contact your Galaxy server administrator.</p>
<h3 id="example-2-convert-fastq-to-fasta">Example 2: Convert Fastq to Fasta</h3>
<p>This shows how to convert a fastq file to a fasta file. The tool creates a new file with the converted data.</p>
<p><strong>Converter tool</strong></p>
<ul>
<li>From the tool panel, click on <strong>Convert Formats -&gt; FASTQ to FASTA</strong> and set the following:</li>
<li>"FASTQ file to convert": <em>Typical Fastq File</em></li>
<li>Click <strong>Execute</strong></li>
</ul>
<p>This will have created a new Fasta file called FASTQ to FASTA on data 2.</p>
<h3 id="example-3-find-ribosomal-rna-features-in-a-dna-sequence">Example 3: Find Ribosomal RNA Features in a DNA Sequence</h3>
<p>This example shows how to use a tool called “barrnap” to search for rRNAs in a DNA sequence.</p>
<p><strong>1. Find all of the ribosomal RNA's in a sequence</strong></p>
<ul>
<li>From the tool panel, click on <strong>Annotation -&gt; barrnap</strong> and set the following:</li>
<li>"Fasta file": MRSA252.fna</li>
<li>Click <strong>Execute</strong></li>
</ul>
<p>A new file called <em>barrnap on data 3</em> will be produced. It is a gff3 file. (This stands for genome feature format - version 3. It is a file format for describing features contained by a DNA sequence.) Change it’s name to something more appropriate (click on the <img src="../media/Galaxy-edit.png" width=20 /> icon.) There is also a STDERR output file from this tool - just ignore this one.</p>
<p>Now lets say you only want the lines of the file for the 23S rRNA annotations. We can do this using a Filter tool.</p>
<p><strong>2. Filter the annotations to get the 23S RNAs</strong></p>
<ul>
<li>From the tool panel, click on <strong>Filter and Sort -&gt; Select</strong> and set the following:</li>
<li>"Select lines from":  (whatever you called the barrnap gff3 output)</li>
<li>"the pattern": <em>23S</em>     (this will look for all the lines in the file that contain “23S”)</li>
<li>Click <strong>Execute</strong></li>
</ul>
<p>Now you have a gff3 file with just the 23S annotations!</p>
<h2 id="what-now">What now?</h2>
<p>Remember how we started a new history at the beginning? If you want to see any of your old histories, click on the history menu button <img src="../media/Galaxy-menu.png" width=20 /> at the top of the histories panel and then select “Saved Histories.” This will give you a list of all the histories you have worked on in this Galaxy server.</p>
<p>That's it. You now know a bit about the Galaxy interface and how to load data, run tools and view their outputs. For more tutorials, see <a href="http://genome.edu.au/learn">http://genome.edu.au/learn</a></p>
{% endblock %}

{% block modals %}
{% endblock %}

{% block scripts %}
{% endblock %}